File: app/src/main/java/com/mss/imagesearcher/domain/repositories/FilesRepository.java
package com.mss.imagesearcher.domain.repositories;

import android.graphics.Bitmap;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.domain.models.ImageInfo;

import java.io.IOException;

public interface FilesRepository {

    @NonNull
    String saveBitmap(@NonNull final ImageInfo imageInfo, @NonNull final Bitmap bitmap) throws IOException;
}

File: app/src/main/java/com/mss/imagesearcher/domain/repositories/FilesRepository.kt
package com.mss.imagesearcher.domain.repositories

import android.graphics.Bitmap

import com.mss.imagesearcher.domain.models.ImageInfo

import java.io.IOException

interface FilesRepository {

    @Throws(IOException::class)
    fun saveBitmap(imageInfo: ImageInfo, bitmap: Bitmap): String
}
